xds/src/main/java/io/grpc/xds/PriorityLoadBalancer.java

```diff
@@ -287,7 +280,12 @@ public void updateBalancingState(ConnectivityState newState, SubchannelPicker ne
             failOverTimer.cancel();
           }
         }
-        tryNextPriority(true);
+        syncContext.execute(new Runnable() {
```

Comment:
I would enqueue `connectivityState = newState; picker = newPicker` together with `tryNextPriority()`. The state update should be "atomic". Same for other balancers.

Reply:
Fixed.